src/parser/pegen.ts

```diff
@@ -467,6 +469,21 @@ export function get_expr_name(e: expr): string {
 //     return NULL;
 // }
 
+export function raise_error_known_location(
+    p: Parser,
+    errtype: typeof pySyntaxError,
+    lineno: number,
+    col_offset: number,
+    msg: string
+): never {
+    if (p.start_rule === FSTRING_INPUT) {
+        /** @todo */
+    }
+    /** @todo should we just set the error indicator and return null then check this in the memoize decorator? */
+    // p.error_indicator = 1;
+    throw new errtype(msg, ["<file>", lineno, col_offset, p.peek().string]);
+}
+
```

Comment:
initial implementation - as above -maybe this should live in the Parser class rather than in pegen.

Reply:
does it live in pegen in cpython?

Reply:
it does, but then so do a lot of things like - `this.expect` and `this.number`...